jskit/encoders/bi.py

```diff
@@ -30,184 +31,249 @@ def set_seed(seed):
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def get_train_config():
+    try:
+        with open("utils/train_config.json", "r") as jsonfile:
+            data = json.load(jsonfile)
+            print("Read successful")
+        return data
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
 
 
-@jaseci_action(act_group=['bi_enc'], allow_remote=True)
-def set_config(training_parameters: Dict = None, model_parameters: Dict = None):
-    """
-    Update the configuration file with any new incoming parameters
-    """
-    global config, save_restart
-    config.read('utils/config.cfg')
-    if training_parameters:
-        config['TRAIN_PARAMETERS'].update(training_parameters)
-    if model_parameters:
-        config['MODEL_PARAMETERS'].update(model_parameters)
-        save_restart = True
-    with open("utils/config.cfg", 'w') as configfile:
-        config.write(configfile)
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def set_train_config(training_parameters: Dict = None):
+    global train_config
+    try:
+        with open("utils/train_config.json", "w+") as jsonfile:
+            train_config.update(training_parameters)
+            json.dump(train_config, jsonfile, indent=4)
+        return "Config setup is complete."
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def get_model_config():
+    try:
+        with open("utils/model_config.json", "r") as jsonfile:
+            data = json.load(jsonfile)
+            print("Read successful")
+        return data
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
+def set_model_config(model_parameters: Dict = None):
+    global model_config
     try:
-        config_setup()
+        with open("utils/model_config.json", "w+") as jsonfile:
+            model_config.update(model_parameters)
+            json.dump(model_config, jsonfile, indent=4)
+
+        config_setup(save_restart=True)
+        return "Config setup is complete."
+
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
-    return "Config setup is complete."
 
 
-@jaseci_action(act_group=['bi_enc'], allow_remote=True)
+@ jaseci_action(act_group=['bi_enc'], allow_remote=True)
 def save_model(model_path: str):
     """
     saves the model to the provided model_path
     """
-    global model, tokenizer, shared, config
+    # global model, tokenizer, model_config
     try:
-        if not model_path.isalnum():
+        if not model_path.replace('_', '').isalnum():
             raise HTTPException(
                 status_code=400,
-                detail='Invalid model name. Only alphanumeric chars allowed.'
+                detail="Invalid model name. Only alphanumeric chars allowed "
```

Comment:
Update this error message to be more accurate -- from the code, it looks like we allow alphanumeric and underscore.